dotCMS/src/main/java/com/dotcms/system/event/local/type/publish/PublishEvent.java

```diff
@@ -0,0 +1,50 @@
+package com.dotcms.system.event.local.type.publish;
+
+import com.dotcms.publisher.business.PublishQueueElement;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+/**
+ * Object used to represent an event to be triggered among a push publishing process, ie: assets
+ * added to queue, push publishing starts, push publishing ends, etc
+ *
+ * @author nollymar
+ */
+public class PublishEvent {
+
+    private String name = null;
+    private List<PublishQueueElement> publishQueueElements = null;
+    private LocalDateTime date;
+
+    public PublishEvent(String name,
+            List<PublishQueueElement> publishQueueElements, LocalDateTime date) {
+        this.name = name;
+        this.publishQueueElements = publishQueueElements;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Store a copy of "publishQueueElements". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2384)